### PR TITLE
Enable shapes layer (right click) > convert to labels

### LINKED ISSUE #3976: Enable shapes layer (right click) > convert to labels
## 🚀 Feature

Enable shapes layer (right click) > convert to labels so that this function is accessible from GUI directly.

## Pitch

Currently it is accessible through API to generate labels layer from shapes layer through napari.layers.Shapes.to_labels, but the option to access from the shapes layer (right click) > convert to labels is grayed out. 
The additional input that might be required is the labels layer data shape. One possible implementation is after clicking on “convert to labels”, add a dialog box to select “match dimension to [available image layer]”, which might be the most common usage. 

## Alternatives

Continue to only support via API. Though it would be nice to have a tooltip when hovering on the grayed out option that shows the corresponding API. 

## Additional context
In general, the right click menu from different layers shows the same contents but grayed out with different parts. 
This could be confusing as the grayed out options may be due to:
(1) Non-applicable actions
(2) Not sufficient selection/condition (i.e. Link Layers is only available if more than one layer is selected)
Ideally (1) should be removed so that users do not have to guess if those actions could be activated at all.

### PR DESCRIPTION
**Pull request by tlambert03**
# Description
closes #3976

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# References

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/docs/dev/guides/translations.html).

### CODE REVIEW

File: napari/layers/_layer_actions.py [hunk 1]
```diff
@@ -107,12 +107,17 @@ def _convert_dtype(ll: LayerList, mode='int64'):
 
 
 def _convert(ll: LayerList, type_: str):
+    from ..layers import Shapes
```
Comment (nclack) [inline napari/layers/_layer_actions.py:110]: is there a reason the import is local?

**Review by nclack — APPROVED** [review]
This works really well in my hands. Love seeing the context keys.

**Comment by chili-chiu** [comment]
Works well for me too! Just a note on its behavior, the generated labels layer has the size starting from 0 to the max of shapes' extent. For example, adding shapes on z = 3 would generate a new labels layer with 4 z slices.

**Review by chili-chiu — APPROVED** [review]

**Comment by Carreau** [comment]
Rebased to fix the `channel_axis` stuff.

**Pull request merged**